Source/carthage/CopyFrameworks.swift

```diff
@@ -101,13 +101,29 @@ private func copyBCSymbolMapsForFramework(_ frameworkURL: URL, fromDirectory dir
 }
 
 private func codeSigningIdentity() -> SignalProducer<String?, CarthageError> {
-	return SignalProducer { () -> Result<String?, CarthageError> in
-		if codeSigningAllowed() {
-			return getEnvironmentVariable("EXPANDED_CODE_SIGN_IDENTITY").map { $0.isEmpty ? nil : $0 }
-		} else {
-			return .success(nil)
+	return SignalProducer(codeSigningAllowed)
+		.attemptMap { codeSigningAllowed in
+			guard codeSigningAllowed == true else { return .success(nil) }
+
+			return getEnvironmentVariable("EXPANDED_CODE_SIGN_IDENTITY")
+				.map { $0.isEmpty ? nil : $0 }
+				.flatMapError {
+					// See https://github.com/Carthage/Carthage/issues/2472#issuecomment-395134166 regarding Xcode 10 betas
+					// … or potentially non-beta Xcode releases of major version 10 or later.
+
+					switch getEnvironmentVariable("XCODE_PRODUCT_BUILD_VERSION") {
+					case .success(_):
```

Comment:
BTW, I don't know if this is some particular Carthage style you're following here, but in general you can omit the `(_)` and just say `case .success:` if you don't want any of the associated values.